mythril_core/src/device/mod.rs

```diff
@@ -118,18 +118,25 @@ impl DeviceMap {
         op.find_device_mut(self)
     }
 
-    //TODO: detect conflicts
     pub fn register_device(&mut self, dev: Box<dyn EmulatedDevice>) -> Result<()> {
         let services = dev.services();
         let dev = Rc::new(dev);
         for region in services.into_iter() {
             match region {
                 DeviceRegion::PortIo(val) => {
                     let key = PortIoRegion(val);
+                    if self.portio_map.contains_key(&key) {
+                        return Err(Error::InvalidDevice("I/O Port already registered!".into()));
```

Comment:
Maybe a little less enthusiasm here 😄. Also, I suspect it may be non-trivial, but if we can derive Debug for PortioRegion and MemIoRegion, then we could include them in this error pretty easily.

Reply:
looks like they derive Debug without a hitch. I've added the conflicting region to the error message. Do you think that it should also include the region it conflicts with?

Reply:
Yes, I think that's reasonable. If you look at the output and it isn't too busy at least